### ISSUE TITLE
Suggestion 1

### ISSUE THREAD

**Issue by SeptyIsHere** [BentoBoxWorld/Boxed#21]
#### Is your feature request related to a problem?

Advancements Customrewards, example: 
You completed the advancement with crafting a diamond armor, you get an animal spawned on your box (or an item of the owner's choise) 

#### Describe the solution you'd like us to implement.

Advancementsrewards.yml here you'll find every advancement that's enabled.

#### Describe alternatives you've considered.

I can't

### Additional context

Like the real CaptiveMinecraft (box)

**Comment by BONNe** [comment]
(the idea is to create a separate addon that does it, but currently noone has a time for that)

**Comment by SeptyIsHere** [comment]
> (the idea is to create a separate addon that does it, but currently noone has a time for that)

I mean an addon should be okay as well, but implementing this in the main plugin it's the best, in my opinion

**Comment by BONNe** [comment]
No, implementing it into the main plugin would mean that BentoBox core has a function that is not necessary for everyone. The idea of the BentoBox is to keep plugin with only main functions to run a gamemode, while addons allow enhancing each server.

If 'by plugin' you mean boxed gamemode, then it would not allow using the same advancement rewards in other GameModes without duplicating the code. Why SkyBlock should not be able to receive awards for advancements, or caveblock... or skygrid?

**Comment by SeptyIsHere** [comment]
Alright.

**Comment by tastybento** [comment]
I have been thinking of adding a rewards option to the advancements.yml file. Also, the idea is that what I learn from Boxed I plan to put into an addon for other game modes.

**Comment by fredthedoggy** [comment]
This is more of a data pack function, as you can already add functions and commands to execute on advancement complete.
I understand adding it here for simplicity, but it is already possible.
